storage_drivers/solidfire_san.go

```diff
@@ -191,18 +191,45 @@ func (d *SolidfireSANStorageDriver) Validate() error {
 	return nil
 }
 
+// Strip storage prefix name to simple Docker Name
+func (d *SolidfireSANStorageDriver) StripVolumePrefix(name string) string {
+	dockerName := name
+	prefix := string(d.Config.CommonStorageDriverConfig.StoragePrefixRaw)
+	if prefix != "" {
+		dockerName = strings.Replace(name, prefix, "", -1)
+	}
+	return dockerName
+}
+
+// Strip snapshot prefix name to simple Docker Name
+func (d *SolidfireSANStorageDriver) StripSnapshotPrefix(name string) string {
+	dockerName := name
+	prefix := string(d.Config.CommonStorageDriverConfig.SnapshotPrefixRaw)
+	if prefix != "" {
+		dockerName = strings.Replace(name, prefix, "", -1)
+	}
+	return dockerName
+}
+
+// Make SolidFire name
+func MakeSolidFireName(name string) string {
+	return strings.Replace(name, "_", "-", -1)
+}
+
 // Create a SolidFire volume
 func (d *SolidfireSANStorageDriver) Create(name string, opts map[string]string) error {
 	log.Debugf("SolidfireSANStorageDriver#Create(%v)", name)
 
 	var req sfapi.CreateVolumeRequest
 	var qos sfapi.QoS
 	var vsz int64
-	var meta = map[string]string{"platform": "Docker-NDVP"}
+	var dName = d.StripVolumePrefix(name)
```

Comment:
Seems kind of hacky for us to do it this way. Adding the prefix and then stripping it. Maybe we should pass in both forms of the name into the APIs instead?